crypto/kdf/pbkdf2.c

```diff
@@ -209,7 +231,17 @@ static int pkcs5_pbkdf2_alg(const char *pass, size_t passlen,
     HMAC_CTX *hctx_tpl = NULL, *hctx = NULL;
 
     mdlen = EVP_MD_size(digest);
-    if (mdlen < 0)
+    if (mdlen <= 0)
+        return 0;
+
+    /* Check that 112 bits <= keylen < (2^32 - 1) * mdlen */
+    if (((keylen * 8) < KDF_PBKDF2_MIN_KEY_LEN_BITS)
+        || ((keylen / mdlen) >= KDF_PBKDF2_MAX_KEY_LEN_DIGEST_RATIO))
```

Comment:
Should we be also checking the value of `range_checks` here?

Reply:
I did ponder it - But I think not..
If it goes above that range the counter overflows which is very bad.

Reply:
> If it goes above that range the counter overflows which is very bad.

Perhaps a comment stating that would be helpful

Is that also true for the `KDF_PBKDF2_MIN_KEY_LEN_BITS` check?

Reply:
Not sure  - I assume that is similar to the salt check so maybe that one goes in the check - in that case I might rename the range_checks variable :)

Reply:
Comment added and the lower bound check has been moved.